fuzz/fuzz_targets/fuzz_expr.rs

```diff
@@ -0,0 +1,184 @@
+    let mut rng = rand::thread_rng();
+    let ops = ["+", "-", "*", "/", "%", "<", ">", "=", "&", "|"];
+
+    let mut expr = String::new();
+    let mut depth = 0;
+    let mut last_was_operator = false;
+
+    while depth <= max_depth {
+        if last_was_operator || depth == 0 {
+            // Add a number
+            expr.push_str(&rng.gen_range(1..=100).to_string());
+            last_was_operator = false;
+        } else {
+            // 90% chance to add an operator followed by a number
+            if rng.gen_bool(0.9) {
+                let op = *ops.choose(&mut rng).unwrap();
+                expr.push_str(&format!(" {} ", op));
+                last_was_operator = true;
+            }
+            // 10% chance to add a random string (potentially invalid syntax)
+            else {
+                let random_str = generate_random_string(rng.gen_range(1..=10));
+                expr.push_str(&random_str);
+                last_was_operator = false;
+            }
+        }
+        depth += 1;
+    }
+
+    // Ensure the expression ends with a number if it ended with an operator
+    if last_was_operator {
+        expr.push_str(&rng.gen_range(1..=100).to_string());
+    }
+
+    expr
+}
+
+fuzz_target!(|_data: &[u8]| {
+    let mut rng = rand::thread_rng();
+    let expr = generate_expr(rng.gen_range(0..=20));
+    let mut args = vec![OsString::from("expr")];
+    args.extend(expr.split_whitespace().map(OsString::from));
+
+    // Save the original stdout file descriptor
+    let original_stdout_fd = unsafe { dup(STDOUT_FILENO) };
+
+    // Create a pipe to capture stdout
+    let mut pipe_fds = [-1; 2];
+    unsafe { libc::pipe(pipe_fds.as_mut_ptr()) };
+    let uumain_exit_code;
+    {
+        // Redirect stdout to the write end of the pipe
+        unsafe { dup2(pipe_fds[1], STDOUT_FILENO) };
+
+        // Run uumain with the provided arguments
+        uumain_exit_code = uumain(args.clone().into_iter());
+
+        // Restore original stdout
+        unsafe { dup2(original_stdout_fd, STDOUT_FILENO) };
+        unsafe { libc::close(original_stdout_fd) };
+    }
+    // Close the write end of the pipe
+    unsafe { libc::close(pipe_fds[1]) };
+
+    // Read captured output from the read end of the pipe
+    let mut captured_output = Vec::new();
+    let mut read_buffer = [0; 1024];
+    loop {
+        let bytes_read = unsafe {
+            libc::read(
+                pipe_fds[0],
+                read_buffer.as_mut_ptr() as *mut libc::c_void,
+                read_buffer.len(),
+            )
+        };
+        if bytes_read <= 0 {
+            break;
+        }
+        captured_output.extend_from_slice(&read_buffer[..bytes_read as usize]);
+    }
+
+    // Close the read end of the pipe
+    unsafe { libc::close(pipe_fds[0]) };
+
+    // Convert captured output to a string
+    let rust_output = String::from_utf8_lossy(&captured_output)
+        .to_string()
+        .trim()
+        .to_owned();
+
```

Comment:
Lines 111 - 156 also look like a candidate for `fuzz_common.rs`, it seems to be identical to the code in `fuzz_test.rs`.